telematic_system/scripts/log_analysis/parse_processing_service_logs.py

```diff
@@ -0,0 +1,82 @@
+import re
+import sys
+
+from enum import Enum
+import datetime
+import pytz
+import numpy as np
+
+class ProcessingStatus(Enum):
+    IN_PROGRESS= "IN_PROGRESS"
+    COMPLETED= "COMPLETED"
+    ERROR= "ERROR"
+
+
+def to_utc_datetime(self, timestamp):
+        return datetime.datetime.fromtimestamp(timestamp).astimezone(pytz.utc).replace(tzinfo=None)
+
+def parse_start_end_processing_line(line, search_str):
+    # Regex match to extract timestamp of the form: "log":"2024-07-23 15:43:45,821
+    timestamp_match = re.search(r'"log":"(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d+)', line)
+    if timestamp_match:
+        timestamp_str = timestamp_match.group(1)
+    else:
+        print("Could not find processing starting time for rosbag")
```

Comment:
If there is no timestamp_str, do we still continue the rest operations?

Reply:
The timestamp is critical to the metrics being calculated in this script, if we cannot find it we should be exiting. I will update this to exit under else condition